gbpservice/nfp/lifecycle_manager/modules/device_lifecycle_manager.py

```diff
@@ -0,0 +1,763 @@
+    device_lifecycle_mgr_obj = DeviceLifeCycleManager(controller, config)
+    evs = [
+        Event(id='CREATE_NETWORK_FUNCTION_DEVICE',
+              handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_SPAWNING', handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_UP', handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_HEALTHY', handler=device_lifecycle_mgr_obj),
+        Event(id='CONFIGURE_DEVICE', handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_CONFIGURED', handler=device_lifecycle_mgr_obj),
+
+        Event(id='DELETE_NETWORK_FUNCTION_DEVICE',
+              handler=device_lifecycle_mgr_obj),
+        Event(id='DELETE_CONFIGURATION', handler=device_lifecycle_mgr_obj),
+        Event(id='DELETE_CONFIGURATION_COMPLETED',
+              handler=device_lifecycle_mgr_obj),
+        Event(id='DELETE_DEVICE', handler=device_lifecycle_mgr_obj),
+
+        Event(id='DEVICE_NOT_UP', handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_NOT_REACHABLE', handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_CONFIGURATION_FAILED',
+              handler=device_lifecycle_mgr_obj),
+        Event(id='DEVICE_ERROR', handler=device_lifecycle_mgr_obj),
+        ]
+    controller.register_events(evs)
+
+
+def module_init(controller, config):
+    events_init(controller, config)
+    rpc_init(controller, config)
+    LOG.info(_("DLCM: module_init"))
+
+
+class RpcHandler(object):
+    RPC_API_VERSION = '1.0'
+
+    def __init__(self, conf, controller):
+        super(RpcHandler, self).__init__()
+        self.conf = conf
+        self._controller = controller
+        self.rpc_event_mapping = {'healthmonitor': ['DEVICE_HEALTHY',
+                                                   'DEVICE_NOT_REACHABLE'],
+                                  'interfaces':    ['DEVICE_CONFIGURED',
+                                                   'DEVICE_CONFIGURED_FAILED'],
+                                  'routes':        ['DEVICE_CONFIGURED',
+                                                   'DEVICE_CONFIGURED_FAILED'],
+                                  }
+
+    # RPC APIs status notification from Configurator
+    def get_network_function_config_info(self, context, notifications_data):
+        #context = kwargs.get('context')
+        #notification_data = kwargs.get('notification_data')
+        responses = notifications_data[0].get('kwargs')
+
+        for response in responses:
+            resource = response.get('resource')
+            request_info = response.get('request_info')
+            result = response.get('result')
+
+            event_id = self.rpc_event_mapping[resource][0]
+            if result != 'success':
+                event_id = self.rpc_event_mapping[resource][1]
+                break
+
+        event_data = request_info
+        self._create_event(event_id=event_id,
+                           event_data=event_data)
+
+
+class DeviceLifeCycleManager(object):
```

Comment:
Why are these methods not part of DeviceLifeCycleHandler() as in the case of service_lifecylce_manager.py?